### DISCUSSION TITLE
Should instrumented tests be executable?

### DISCUSSION THREAD

**Discussion by beraldoleal** [avocado-framework/avocado#4380] [RFC]
This was pointed by Max Reitz:

> For example, something like this:
> 
> test.py:
> 
>     #!/usr/bin/python
>     :
> 
> Run with:
> 
>   $ avocado run test.py
> 
> And now I can see that if test.py is executable, I indeed get:
> 
>   ...
>    (1/1) test.py: FAIL: Exited with status: '1', stdout: '' stderr: ' File
> "/tmp/test.py", line 13\n    :\n    ^\nSyntaxError: invalid syntax\n' (0.06
> s)
> 
> However, if test.py is not executable, I get the “Unable to resolve
> reference(s) 'test.py' with plugin(s)...” error I noted above.  Perhaps a
> shebang line indeed doesn’t make sense without the file being executable.
> OTOH, I don’t think there’s a reason for making a Python script executable
> if it doesn’t contain top-level code, which is why I never did for tests.

What do you think about this?

**Comment by pevogam** [comment]
I think it is a good idea to keep the shebang only for directly invoked python scripts. OTOH one could consider `avocado run` as invocation of the script so this is indeed quite relative.
